fe/fe-core/src/main/java/com/starrocks/transaction/LakeTableTxnLogApplier.java

```diff
@@ -90,4 +94,55 @@ public void applyVisibleLog(TransactionState txnState, TableCommitInfo commitInf
             }
         }
     }
+
+    public void applyVisibleLogBatch(TransactionStateBatch txnStateBatch, Database db) {
+        for (TransactionState txnState : txnStateBatch.getTransactionStates()) {
+            TableCommitInfo commitInfo = txnState.getTableCommitInfo(txnStateBatch.getTableId());
```

Comment:
The difference is that the partitionCommitInfo in each transactionState does not have compactionScore information and needs to be obtained from `StateBatch`. I will  assgin compactionScore to each transactionState after publish successfully so that can call `applyVisibleLog()` directly